### PR TITLE
[IMP] Adding purge toolbar button in overrides manager(solves #6816)

### PR DESCRIPTION
**Pull request by infograf768**
See https://github.com/joomla/joomla-cms/issues/6816

This patch adds a Purge button in languages/overrides => Manager.
screen shot 2015-04-23 at 18 26 26

### CODE REVIEW

**Comment by coolman01** [comment]
Tested and functions as described.
Thank you

File: administrator/components/com_languages/models/overrides.php [hunk 1]
```diff
@@ -286,4 +286,30 @@ public function delete($cids)
...
+	 */
+	public function purge()
+	{
+		$db = JFactory::getDbo();
+
+		// Note: TRUNCATE is a DDL operation
+		// This may or may not mean depending on your database
+		try
+		{
+			$db->setQuery('TRUNCATE TABLE #__overrider');
```
Comment (wilsonge) [inline administrator/components/com_languages/models/overrides.php]: @infograf768 We should use the built in ```truncateTable``` in JDatabase https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/database/driver.php#L1779

**Comment by JoshuaLewis** [comment]
Just trying to better understand the purge function, so it's usage is mostly for caching situations with modifications?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6827">issues.joomla.org/joomla-cms/6827</a>.</sub>

**Comment by infograf768** [comment]
@wilsonge 
truncateTable implemented. Thanks

@JoshuaLewis 
The overrider table is ONLY a cache table storing the strings for the language targeted (it does not contain the overrides created in language/overrides/xx-XX.override.ini) . 
It is deleted and refilled <b>when clicking in the search field.</b>
See method refresh() 
https://github.com/joomla/joomla-cms/blob/staging/administrator/components/com_languages/models/strings.php#L26

What happens is that this table can become very heavy if overrides were created for many languages. 
When moving the site and database, the table contains wrong path for the files containing the language strings. They are normally refreshed the first time one clicks in the search field.

I could not reproduce here the issue described by @coolman01 but it is not a big deal to add a Purge button for that table for people who have issues.

**Comment by brianteeman** [comment]
I know the table is called Overrider but thats not really English so can you change the message to use either the word override or overrides.

**Comment by coolman01** [comment]
Thanks guys, especially @infograf768 for your support here.

**Comment by infograf768** [comment]
>I know the table is called Overrider but thats not really English so can you change the message to use either the word override or overrides. 

Shall we let CMS maintainers decide about that? The table is called "overrider" and does not contain any override. Changing the name would create a confusion as it could be understood that the table contains the overrides.
Coca-Cola or Joomla! are not English either...

**Comment by brianteeman** [comment]
Fair enough and I see that the word table is in the string so I guess it is
OK (I was checking on my phone before which isn't so easy)

Please add a period at the end of the string to be in line with the ;en-GB
language style guide

On 24 April 2015 at 11:49, infograf768 <notifications@github.com> wrote:

> I know the table is called Overrider but thats not really English so can
> you change the message to use either the word override or overrides.
>
> Shall we let CMS maintainers decide about that? The table is called
> "overrider" and does not contain any override. Changing the name would
> create a confusion as it could be understood that the table contains the
> overrides.
> Coca-Cola or Joomla! are not English either...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/pull/6827#issuecomment-95890633>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by infograf768** [comment]
Period added.

**Comment by brianteeman** [comment]
Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6827">issues.joomla.org/joomla-cms/6827</a>.</sub>

**Comment by zero-24** [comment]
@test successful thanks :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6827">issues.joomla.org/joomla-cms/6827</a>.</sub>

**Comment by zero-24** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6827">issues.joomla.org/joomla-cms/6827</a>.</sub>

**Comment by wilsonge** [comment]
Merged into 3.5-dev with https://github.com/joomla/joomla-cms/commit/fcac315b6aac6845105ec598a8998da7e690aea6

**Pull request closed without merging**